src/Agent.Listener/Capabilities/AgentCapabilitiesProvider.cs

```diff
@@ -34,6 +35,10 @@ public Task<List<Capability>> GetCapabilitiesAsync(AgentSettings settings, Cance
 #if OS_WINDOWS
             Add(capabilities, "Agent.OSVersion", GetOSVersionString());
             Add(capabilities, "Cmd", Environment.GetEnvironmentVariable("comspec"));
+
+            //This capability gets set irrespective of agent being run in auto-logon mode or not.
+            var configurationManager = HostContext.GetService<IConfigurationManager>();
+            Add(capabilities, "InteractiveSession", (!configurationManager.IsServiceConfigured()).ToString());
```

Comment:
1. This seems to be a corner/non-default scenario and we shouldn’t bother about this. In such cases the user should follow the right course for the things to work correctly. 
2. Today Agent.Listener.exe doesn’t know about its origin (process/service), in both cases we are calling “agent.listner.exe run”.


---
In reply to: [118030054](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118030054) [](ancestors = 118030054)